crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -525,7 +525,7 @@ impl SsaContext {
         //Add a new instruction to the nodes arena
         let mut i = Instruction::new(opcode, optype, Some(self.current_block));
         //Basic simplification
-        optim::simplify(self, &mut i)?;
+        optim::simplify(self, &mut i).err();
```

Comment:
Why was this changed to no longer forward the error?

Reply:
That's the main point of the PR; to not send runtime errors until the IF instructions are processed.
Here the function is used to generate the SSA code when processing the AST so we cannot have processed the IF instructions.

Reply:
Wouldn't this mean we throw out any errors that may be found before then? E.g. if a constraint is always false?

Perhaps it is expected when we pass through this block again afterward we will regenerate the same error but I wonder if there are cases where this may not be true. E.g. if there are errors that can happen for only these passes or if an instruction generates an error but is deleted and not reviewed again.

Reply:
The different passes should not change the behavior of the code (unless there is a bug), so the false constraint will be fired eventually later on. Even if it is not handled (but it should not happen), it is part of the instruction set (again unless there is a bug) and will fail in the output circuit, so I think we are safe.

Reply:
I'm still somewhat hesitant on this but suppose it is fine if it is limited to `optim::simplify` which is called often. I will say lets change it to `.ok()` which is somewhat more common for ignoring errors, and lets add a comment for each case briefly explaining/justifying why we're ignoring errors in these instances.

Reply:
Done